File: leetcode/solutions/0961.N-RepeatedElementInSize2NArray/repeatedNTimes_test.go
package solution

import (
	"testing"

	"github.com/google/go-cmp/cmp"
)

func Test_0961_NRepeatedElementInSize2NArray(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  []int
		out int
	}{
		{[]int{1, 2, 3, 3}, 3},
		{[]int{2, 1, 2, 5, 3, 2}, 2},
		{[]int{5, 1, 5, 2, 5, 3, 5, 4}, 5},
	}

	for i, tt := range tests {
		i, tt := i, tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := repeatedNTimes(tt.in)

			if !cmp.Equal(got, tt.out) {
				t.Errorf("\n#%02d\ngot: %#v\nwant: %#v", i, got, tt.out)
			}
		})
	}
}

File: leetcode/solutions/0961.N-RepeatedElementInSize2NArray/repeatedNTimes_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import repeatedNTimes from "./repeatedNTimes.ts";

Deno.test("0961. N-Repeated Element in Size 2N Array", () => {
  assertEquals(repeatedNTimes([1, 2, 3, 3]), 3);
  assertEquals(repeatedNTimes([2, 1, 2, 5, 3, 2]), 2);
  assertEquals(repeatedNTimes([5, 1, 5, 2, 5, 3, 5, 4]), 5);
});
